File: 8kyu/reduce-but-grow/reduce-but-grow.go
package main

import "fmt"

func Grow(arr []int) int {
	prod := 1

	for _, num := range arr {
		prod *= num
	}

	return prod
}

func main() {
	fmt.Println(Grow([]int{1, 2, 3}))    // 6
	fmt.Println(Grow([]int{1, 2, 3, 4})) // 24
}

File: 8kyu/reduce-but-grow/reduce-but-grow.py
import functools
from typing import List

def grow(arr: List[int]) -> int:
    return functools.reduce(lambda x, y: x * y, arr)

# def grow(arr: List[int]) -> int:
#     prod = 1
#     for num in arr:
#         prod *= num
#     return prod



print(grow([1, 2, 3, 4])) #24